### PR TITLE
chore: refactor configuration files and streamline project organization

### PR DESCRIPTION
**Pull request by swiizyy**

### CODE REVIEW

File: .editorconfig
Comment (Arcoz0308) [inline .editorconfig:1]: eslint fix intent size, and is configured to 2 spaces 😄

File: .gitignore [hunk 1]
```diff
@@ -1,6 +1,31 @@
-.idea/
-.env
```
Comment (Arcoz0308) [inline .gitignore:2]: add also .env in net gitignore please

File: .gitattributes
Comment (Arcoz0308) [inline .gitattributes:1]: For what is this ?

File: .vscode/extensions.json
Comment (Arcoz0308) [inline .vscode/extensions.json:1]: i don't use vscode, but cool for other people using vscode, thanks !

File: packages/cli/package.json [hunk 2]
```diff
@@ -31,11 +31,11 @@
...
-  "directories": {
-    "lib": "./src"
-  },
   "dependencies": {
     "commander": "^12.1.0"
   },
-  "type": "module"
+  "type": "module",
+  "files": [
+    "src/"
```
Comment (Arcoz0308) [inline packages/cli/package.json]: output are in dist also

File: .gitattributes
Comment (swiizyy) [reply .gitattributes:1]: To force the use of lf instead of crlf.

There is too much problem between windows and unix users otherwise.

**Pull request merged**